desc/protoparse/linker_test.go

```diff
@@ -906,6 +906,84 @@ func TestLinkerValidation(t *testing.T) {
 			},
 			"foo.proto:4:26: field foobar: option json_name is not allowed on extensions",
 		},
+		{
+			map[string]string{
+				"foo.proto": "syntax = \"proto3\";\n" +
+					"package foo.foo;\n" +
+					"import \"other.proto\";\n" +
+					"service Foo { rpc Bar (Baz) returns (Baz); }\n" +
+					"message Baz {\n" +
+					"  foo.Foo.Bar f = 1;\n" +
+					"}\n",
+				"other.proto": "syntax = \"proto3\";\n" +
+					"package foo;\n" +
+					"message Foo {\n" +
+					"  enum Bar { ZED = 0; }\n" +
+					"}\n",
+			},
+			"foo.proto:6:3: field foo.foo.Baz.f: invalid type: foo.foo.Foo.Bar is a method, not a message or enum",
+		},
+		{
+			map[string]string{
```

Comment:
This test case passed without the scope change (though this example fails with `protoc`).

I added the above test case, too, because I _expected_ it to also incorrectly pass. But, due to some subtleties in the code flow, it was not subject to the same resolution bug. I left it in anyway.

The subsequent two test cases are similar, but are testing resolution of option names in a few different contexts (whereas as these two are testing resolution of field types).